HealthMonitoring.Persistence/EndpointStatsPersistCoordinator.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using HealthMonitoring.Forwarders;
+using HealthMonitoring.Management.Core;
+using HealthMonitoring.Management.Core.Repositories;
+using HealthMonitoring.Model;
+
+namespace HealthMonitoring.Persistence
+{
+    public class EndpointStatsPersistCoordinator : IEndpointStatsPersistCoordinator

```

Comment:
This class looks a bit suspicious.
It is a full proxy on top of the IEndpointStatsRepository but it is designed only to forward new stats.

I think I would remove all methods but InsertEndpointStatistics() from IEndpointStatsPersistCoordinator and let classes using those methods to call IEndpointStatsRepository  directly.